EIPS/eip-4410.md

```diff
@@ -0,0 +1,160 @@
+
+While this standard focuses on NFTs and compatibility with the ERC-721 and ERC-1155 standards, EIP-4410 does not require compatibility with ERC-721 and ERC-1155 standards. Any other contract could integrate with EIP-4410 to describe the memories for any other data structure. ERC-4410 is, therefore, a universal memory standard for many asset types.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
+NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and
+"OPTIONAL" in this document are to be interpreted as described in
+RFC 2119.
+
+**ERC-721 and ERC-1155 compliant contracts MAY implement this ERC to provide a standard method of specifying memory information.**
+
+dApps that support this standard **MUST** implement some method of storing the memory metadata. Standards for specifying this metadata are including in this EIP.
+
+Implementers of this standard **MUST** have all of the following functions:
+
+```solidity
+
+pragma solidity ^0.6.0;
+import "./IERC165.sol";
+
+///
+/// @dev Interface for the NFT Memories Standard
+///
+interface IERC4410 is IERC165 {
+    /// ERC165 bytes to add to interface array - set in parent contract
+    /// implementing this standard
+    ///
+    /// bytes4(keccak256("addMemory(uint256,string)")) == 0x2a55205a
+    /// bytes4 private constant _INTERFACE_ID_ERC4410 = 0x2a55205a;
+    /// _registerInterface(_INTERFACE_ID_ERC4410);
+
+    /// @notice Called with the tokenID and the URI of the memory's metadata
+    /// @param _tokenId - the NFT asset that the memory is added to 
+    /// @param metadataURI - the URI that points to the memory metadata
+    function addMemory(uint256 _tokenId, string calldata _metadataURI) external {
+        require(_exists(_tokenId), "ERC721: operator query for nonexistent token");
+        require(_isApprovedOrOwner(_msgSender(), _itemId), "CONTRACT::addMemory: caller is not owner nor approved");
+        emit LogMemoryEvent(_itemId, _metadataURI);
+    }
+
+    event LogMemoryEvent(uint256 indexed itemId, string metadataURI );
+}
+
+interface IERC165 {
+    /// @notice Query if a contract implements an interface
+    /// @param interfaceID The interface identifier, as specified in ERC-165
+    /// @dev Interface identification is specified in ERC-165.
+    /// @return `true` if the contract implements `interfaceID` and
+    ///  `interfaceID` is not 0xffffffff, `false` otherwise
+    function supportsInterface(bytes4 interfaceID) external view returns (bool);
+}
+```
+
+Implementers of this standard **MUST** implement the metadata using the following JSON schema (note that only tokenId and externalUrl are required):
+
+```yaml
+type: object
+properties:
```

Comment:
If I'm not mistaken (I might be, feel free to correct me), this isn't compliant with OpenSea's [metadata standards](https://docs.opensea.io/docs/metadata-standards). While OS's standard is not an EIP, something you may want to consider is how you think it might affect traction of this EIP for it to not work well with it.

In a similar vein, would it make more sense to move the metadata specific to this EIP into its own object? That would allow you to have a root level that's compliant with marketplaces, and an additional object with the items specific to this EIP.

Reply:
Response to first point:  I believe it is compliant with OpenSea's metadata standards because it is completely independent of OS's metadata standards.  Both can be used in conjunction with each other, but can also be used independently of each other.

Response to second point: Because these specifications are independent of each other, that would not be a reason for move the metadata aspect into its own object.  That said, it may be a good idea to move the metadata aspect into its own object anyway (what is the precedent for that?).